app/engine/utils/OLSLinearSeries.js

```diff
@@ -0,0 +1,203 @@
+'use strict'
+/*
+  Open Rowing Monitor, https://github.com/laberning/openrowingmonitor
+
+  The LinearSeries is a datatype that represents a Linear Series. It allows
+  values to be retrieved (like a FiFo buffer, or Queue) but it also includes
+  a Linear Regressor to determine the slope, intercept and R^2 of this timeseries
+  of x any y coordinates through Simple Linear Regression.
+
+  At creation it can be determined that the Time Series is limited (i.e. after it
+  is filled, the oldest will be pushed out of the queue) or that the the time series
+  is unlimited (will only expand). The latter is activated by calling the creation with
+  an empty argument.
+
+  please note that for unlimited series it is up to the calling function to handle resetting
+  the Linear Series when needed through the reset() call.
+
+  A key constraint is to prevent heavy calculations at the end (due to large
+  array based curve fitting) as this function is also used to calculate
+  drag at the end of the recovery phase, which might happen on a Pi zero
+
+  This implementation uses concepts that are described here:
+  https://www.colorado.edu/amath/sites/default/files/attached-files/ch12_0.pdf
+*/
+
+import { createSeries } from './Series.js'
+
+import loglevel from 'loglevel'
+const log = loglevel.getLogger('RowingEngine')
+
+function createOLSLinearSeries (maxSeriesLength = 0) {
+  const X = createSeries(maxSeriesLength)
+  const XX = createSeries(maxSeriesLength)
+  const Y = createSeries(maxSeriesLength)
+  const YY = createSeries(maxSeriesLength)
+  const XY = createSeries(maxSeriesLength)
+  const trend = createSeries(maxSeriesLength)
+  let _slope = 0
+  let _intercept = 0
+  let _goodnessOfFit = 0
+
+  function push (x, y) {
+    X.push(x)
+    XX.push(x * x)
+    Y.push(y)
+    YY.push(y * y)
+    XY.push(x * y)
+
+    // Let's approximate the line through OLS
+    if (X.length() >= 2 && X.sum() > 0) {
+      _slope = (X.length() * XY.sum() - X.sum() * Y.sum()) / (X.length() * XX.sum() - X.sum() * X.sum())
```

Comment:
First of all, OLS is O(1), so it isn't a huge waist of time. We could make it lazy, but it wouldn't make it much of a difference. On a Pi 4, you wouldn't notice it.

Reply:
I had the Pi Zero (the Zero 2 has only 1Ghz single core) or other equipment with lower CPU in mind. Of course this is just a recommendation and can be seen as micro optimization. This can also be done in a second step after the beta is validated.